src/tests/testCacheManager.cc

```diff
@@ -180,35 +205,45 @@ TestCacheManager::testParseUrl()
         "#fragment"
     };
 
+    const auto &magic = CacheManager::MagicUrlPathPrefix();
+
+    assert(magic.length());
+    const auto insufficientMagic = magic.substr(0, magic.length()-1);
+
     for (const auto &scheme : validSchemes) {
         mgrUrl.setScheme(scheme);
 
-        for (const auto *magic : magicPrefixes) {
-
-            // all schemes require magic path prefix bytes
-            if (strlen(magic) <= 2)
-                continue;
+        // Check that the parser rejects URLs that lack the full magic prefix.
+        // These negative tests log "Squid BUG: assurance failed" ERRORs because
+        // they violate CacheManager::ParseUrl()'s ForSomeCacheManager()
+        // precondition.
+        for (const auto *action : validActions) {
+            for (const auto *param : validParams) {
+                for (const auto *frag : validFragments) {
+                    SBuf bits;
+                    bits.append(insufficientMagic);
+                    bits.append(action);
+                    bits.append(param);
+                    bits.append(frag);
+                    mgrUrl.path(bits);
+                    mgr->testInvalidUrl(mgrUrl, "insufficient magic");
+                }
+            }
+        }
 
+        {
```

Comment:
This is just a diff reducing/clarifying addition. Please let me know if you want me to delete it (and shift the enclosed code left).

Reply:
yes, drop this indentation please.

Reply:
Done in commit f8efc2c.